mailbox/api/src/main/java/org/apache/james/mailbox/indexer/ReIndexer.java

```diff
@@ -29,17 +31,60 @@
 public interface ReIndexer {
 
     class RunningOptions {
-        public static final RunningOptions DEFAULT = new RunningOptions(50);
+        public static class Builder {
+            private Optional<Integer> messagesPerSecond;
+            private Optional<Mode> mode;
+
+            public Builder() {
+                this.messagesPerSecond = Optional.empty();
+                this.mode = Optional.empty();
+            }
+
+            public Builder messagesPerSeconds(Optional<Integer> messagesPerSecond) {
+                this.messagesPerSecond = messagesPerSecond;
+                return this;
+            }
+
+            public Builder mode(Optional<Mode> mode) {
+                this.mode = mode;
+                return this;
+            }
+
+            public RunningOptions build() {
+                return new RunningOptions(
+                    messagesPerSecond.orElse(DEFAULT.getMessagesPerSecond()),
+                    mode.orElse(DEFAULT_MODE)
+                );
+            }
+        }
+
+        public static Builder builder() {
+            return new Builder();
+        }
+
+        public enum Mode {
+            rebuild,
+            correct
```

Comment:
We tend to use capital letters for enum. If you want to accept variations, you might want to write a parse static method.